### DISCUSSION TITLE
Add keyboard toolbar for markdown editors

### DISCUSSION THREAD

**Discussion by theontho** [standardnotes/forum#2720] [Suggestions]
**Have you read the [Longevity Statement](https://standardnotes.org/longevity)?** yes

**Is your feature request related to a problem? Please describe.**
**Describe the solution you'd like**
**Describe alternatives you've considered**
When you edit markdown files, you need to enter special characters like `#`.  A keyboard toolbar that shows up above your keyboard like other mobile markdown editors (ex: inkdrop) would make writing markdown a lot nicer then having to tap twice to switch to the second alternative keyboard to reach the `#` character.     It should be an available option for any markdown or editor extension.   Maybe they can specify a 'toolbar characters' array in some json definition somewhere, so the code editor can switch it as you switch languages and so on.  C & Java would put `{`, `}`, `;`, `<`. Markdown `#`, `>`, `*`, `=`, `-` and so on.

**Comment by JaspalSuri** [comment]
Hi @theontho,

The [Markdown Pro Editor](https://standardnotes.org/extensions/markdown-pro) provides a WYSIWYG interface with the toolbar that you're requesting (if I'm not mistaken). 
![image](https://user-images.githubusercontent.com/44656000/100419429-63c42480-3039-11eb-85f4-487b7aa490fd.png)

Feel free to try it out [here](https://standardnotes.org/extensions/markdown-pro). 🙂

**Comment by theontho** [comment]
This one is hovering at the top and goes away. What I'm talking about is one that sticks to the keyboard.  This is the code for it in iOS: https://stackoverflow.com/questions/23904848/how-can-i-add-a-toolbar-above-the-keyboard  It's called the inputAccessoryView

**Comment by theontho** [comment]
Here is an example from Inkdrop 
![image](https://user-images.githubusercontent.com/22130/100550860-b0725000-3231-11eb-866a-25c60db2e11c.png)
